mock/service/controller.go

```diff
@@ -159,6 +159,18 @@ func (s *service) ControllerUnpublishVolume(
 	req *csi.ControllerUnpublishVolumeRequest) (
 	*csi.ControllerUnpublishVolumeResponse, error) {
 
+	if len(req.VolumeId) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID cannot be empty")
+	}
+	if len(req.NodeId) == 0 {
+		// If node id is empty, no failure as per Spec
+		return &csi.ControllerUnpublishVolumeResponse{}, nil
+	}
+
+	if req.NodeId != s.nodeID {
+		return nil, status.Errorf(codes.NotFound, "Not matching Node ID %s to Mock Node ID %s", req.NodeId, s.nodeID)
```

Comment:
remove the `Mock` string. When it is normally used it will be against a real CSI Driver..

Reply:
good catch will do.